track_status_request_map.go

```diff
@@ -1,39 +1,31 @@
 package moqtransport
 
 import (
-	"strings"
 	"sync"
 )
 
 type trackStatusRequestMap struct {
 	lock     sync.Mutex
-	requests map[string]*trackStatusRequest
+	requests map[uint64]*trackStatusRequest
 }
 
 func newTrackStatusRequestMap() *trackStatusRequestMap {
 	return &trackStatusRequestMap{
 		lock:     sync.Mutex{},
-		requests: map[string]*trackStatusRequest{},
+		requests: map[uint64]*trackStatusRequest{},
 	}
 }
 
 func (m *trackStatusRequestMap) add(tsr *trackStatusRequest) bool {
 	m.lock.Lock()
 	defer m.lock.Unlock()
-	// TODO: Implement better mapping than just concatenating all namespace
-	// elements and trackname?
-	key := strings.Join(append(tsr.namespace, tsr.trackname), "")
-	if _, ok := m.requests[key]; ok {
-		return false
-	}
-	m.requests[key] = tsr
+	m.requests[tsr.requestID] = tsr
 	return true
 }
 
-func (m *trackStatusRequestMap) delete(namespace []string, trackname string) (*trackStatusRequest, bool) {
+func (m *trackStatusRequestMap) delete(rid uint64) (*trackStatusRequest, bool) {
```

Comment:
I think it is clearer to keep a longer name for the parameter like "requestID" (applies to other functions as well)